exporter/tanzuobservabilityexporter/transformer.go

```diff
@@ -213,32 +209,26 @@ func attributesToTags(attributes ...pcommon.Map) map[string]string {
 			return true
 		})
 	}
-	return tags
-}
-
-func appAttributesToTags(attributes pcommon.Map) map[string]string {
-	tags := map[string]string{}
 
 	// Resource attr `service` will take preference over `service.name`
-	if resAttrVal, ok := attributes.Get(labelService); ok {
-		tags[labelService] = resAttrVal.AsString()
+	if _, ok := tags[labelService]; !ok {
+		if svcName, svcNameOk := tags[conventions.AttributeServiceName]; svcNameOk {
+			tags[labelService] = svcName
+			delete(tags, conventions.AttributeServiceName)
+		}
 	}
+	return tags
+}
 
+func appAttributesToTags(attributes pcommon.Map) map[string]string {
+	tags := attributesToTags(attributes)
```

Comment:
I like this code you wrote. It is simple and to the point.  Unfortunately, the call to attributesToTags() will make this call run in O(n) time even though you only care about pulling four attributes.  

Consider extracting a function from the code you wrote from lines 214-219 which makes `service` takes preference over `service.name` call it something like ```func fixService(attrs map[string]string)``` fixService can modify the map it is given in place. Then in ```appAttributesToTags()``` method you can just cherry pick the four attributes you want from the `attributes` map the way you did before which will take O(1) time and then call `fixService` on the resulting map which will also take O(1) time.  If I am oversimplifying and my proposal won't work let me know.

Reply:
Thank you, Travis. I have kept the original attributeToTags() method as is and created two new methods fixServiceTags() and appAttributesToTags(). In the appAttributesToTags() method I am not iterating through the whole map and just getting `application`, `service.name`, `service`, `shard`, and `cluster`. And then we call fixServiceTags() if both the `service` and `service.name` are provided. I think this will make it more readable. Let me know if this looks good.